### PR TITLE
Add GitHub Actions job to set milestone for merged PR

### PR DESCRIPTION
**Pull request by ebyhr**
Here is my sample repository having the same job: https://github.com/ebyhr/github-actions-playground

### CODE REVIEW

**Review by electrum — APPROVED** [review]
Nice!

File: .github/workflows/milestone.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  set-milestone:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Get milestone from pom.xml
+        run: |
+          echo "MILESTONE_NUMBER=$(./mvnw -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec | cut -d- -f1)" >> $GITHUB_ENV
```
Comment (electrum) [inline .github/workflows/milestone.yml:17]: Nit: could shorten `--non-recursive` to `-N`

We do this a different way in a couple other places:
```
./mvnw -q help:evaluate -Dexpression=project.version -DforceStdout
```

But the `exec:exec` way is about **1.0s** vs **1.6s** for `help:evaluate` on my machine, so we could switch everywhere.

Comment (ebyhr) [reply .github/workflows/milestone.yml]: Exactly, `exec:exec` is faster than `help:evaluate` on my machine too. Let me handle in follow-up PR.

**Comment by ebyhr** [comment]
The job looks failed. I'm debugging now. 
Screen Shot 2021-09-14 at 11 51 39

**Comment by electrum** [comment]
I think the dots are from `mvnw` downloading Maven the first time. The wrapper code [logs to stdout](https://github.com/takari/maven-wrapper/blob/master/src/main/java/org/apache/maven/wrapper/Logger.java) which is not nice behavior. An easy fix is to run it first with `./mvnw -v` which will force a download.

**Comment by electrum** [comment]
cc @mosabua it would be nice if Maven wrapper logged to stderr instead of stdout

**Pull request merged**